distributed/deploy/spec.py

```diff
@@ -216,9 +227,10 @@ def __await__(self):
     async def _wait_for_workers(self):
         # TODO: this function needs to query scheduler and worker state
         # remotely without assuming that they are local
```

Comment:
I should probably remove this TODO.  I think that now that we're accessing the comm rather than the scheduler that we're handling the remote case.

Reply:
I seem to still be getting caught in this loop forever despite it using the `scheduler_comm`.  Only when starting a cluster with workers and awaiting the cluster object. When scaling up it's fine.

Reply:
Perhaps there is some inconsistency between `self.workers` (the cluster's expected worker set) and the return value of `identity` (what the scheduler sees)?  It might be worth printing out what these look like to see if, for example, they are using different naming conventions.

Reply:
Weirdly it seems to be that `await cluster.scheduler_comm.identity()` hangs forever every other time you call it. If I keyboard interrupt it and run it again I get the output twice.

Reply:
That seems odd.  Do you have a branch where I can try this out?

Reply:
Happening for me on master of [dask-cloud](https://github.com/jacobtomlinson/dask-cloud). 

```python
In [1]: from dask_cloud import ECSCluster

In [2]: cluster = ECSCluster()

In [3]: await cluster.scheduler_comm.identity()  # hangs, press ctrl+c

In [4]: await cluster.scheduler_comm.identity()  # Returns
```

I'll look at working up a minimum reproducible example if I keep getting stuck with this.